packages/cozy-pouch-link/src/helpers.spec.js

```diff
@@ -1,32 +1,59 @@
-import {
-  withoutDesignDocuments,
-  isDesignDocument,
-  isDeletedDocument
-} from './helpers.js'
+import { find, isDesignDocument, isDeletedDocument } from './helpers.js'
+import PouchDB from 'pouchdb'
+import PouchDBFind from 'pouchdb-find'
+import adapter from 'pouchdb-adapter-memory'
+PouchDB.plugin(PouchDBFind)
+PouchDB.plugin(adapter)
+
+const insertData = async (db, number = 2000) => {
+  for (let i = 0, time = new Date().getTime(); i < number; i++, time--) {
+    const doc = { _id: `doc${i}` }
+    try {
+      await db.put(doc)
+    } catch (e) {
+      console.log(`error(${doc._id}): ${e}`)
+    }
+  }
+}
 
 describe('Helpers', () => {
-  describe('withoutDesignDocuments', () => {
-    let response
-    beforeEach(() => {
-      response = {
-        offset: 0,
-        rows: [{ id: 'goodId' }, { id: '_design/wrongId' }],
-        total_rows: 2
-      }
-    })
-    it('should remove design document', () => {
-      const filteredResponse = withoutDesignDocuments(response)
-      expect(filteredResponse.rows.length).toEqual(1)
-      expect(filteredResponse.rows[0].id).toEqual('goodId')
-    })
-    it('should update total rows number', () => {
-      const filteredResponse = withoutDesignDocuments(response)
-      expect(filteredResponse.total_rows).toEqual(1)
-    })
-    it('should not mute response', () => {
-      const responseCopy = { ...response }
-      withoutDesignDocuments(response)
-      expect(response).toEqual(responseCopy)
+  describe('find', () => {
+    let db
+
+    beforeEach(async () => {
+      db = new PouchDB('test', { adapter: 'memory' })
+    })
+
+    afterEach(async () => {
+      await db.destroy()
+    })
+
+    it('should find 200 docs', async () => {
+      await insertData(db, 200)
+      const data = await find(db)
+      expect(data.docs).toHaveLength(200)
+    })
+
+    it('should find 20 docs with limit', async () => {
+      await insertData(db, 200)
+      const data = await find(db, { limit: 20 })
+      expect(data.docs).toHaveLength(20)
+    })
+
+    it('should find 2000 docs', async () => {
+      jest.spyOn(db, 'find')
+      await insertData(db, 2000)
+      const data = await find(db, { sqlite: true })
```

Comment:
Instead of passing `sqlite` as an argument to this function, can we create a `isTheCurrentAdapterIsBugged()` in `src/helpers.js`
 and mock it in this test ? It'll remove the need of the usage of https://github.com/cozy/cozy-client/pull/326/files#diff-50dbbfdf356860744ae7a6b4c5cec814R13

Reply:
`isCurrentAdapterBugged`

Sinon ya deux fois `is` dans le nom de la fonction.